app/views/recipes/_form.html.erb

```diff
@@ -133,4 +133,39 @@
     </div>
   </div>
 
-<% end %>
\ No newline at end of file
+  <div class='form-group input-switch' ng-switch='recipe_<%= @recipe.id %>_file' ng-init='recipe_<%= @recipe.id %>_file=<%= @recipe.file.present? %>'>
```

Comment:
I'm thinking since this is a pretty big form, we can probably put the "Add a link" and the "Upload a File" buttons in the same row. The way to do that would be to put this div (e.g. the recipe file one) and the one above (the recipe link one) into a single bootstrap row with each element taking up six columns. In pseudo code that would be:

```
<div class='row'>
  <div class='col-sm-6'>
    link_div
  </div>
  <div class='col-sm-6'>
    file_div
  </div>
</div>
```

See how this works per the docs here: http://getbootstrap.com/css/#grid-example-mixed